code/game/objects/items/robot/robot_parts.dm

```diff
@@ -62,6 +62,81 @@
+	var/turf/T = get_turf(src)
+	forceMove(T)
+	if(l_leg || r_leg || chest || l_arm || r_arm || head)
+		if(I.use_tool(src, user, 5, volume=50))
+			if(l_leg)
+				l_leg.forceMove(T)
+				l_leg = null
+			if(r_leg)
+				r_leg.forceMove(T)
+				r_leg = null
+			if(chest)
+				if (chest.cell) //Sanity check.
+					chest.cell.forceMove(T)
+					chest.cell = null
+				chest.forceMove(T)
+				new /obj/item/stack/cable_coil(T, 1)
+				chest.wired = FALSE
+				chest = null
+			if(l_arm)
+				l_arm.forceMove(T)
+				l_arm = null
+			if(r_arm)
+				r_arm.forceMove(T)
+				r_arm = null
+			if(head)
+				head.forceMove(T)
+				head.flash1.forceMove(T)
+				head.flash1 = null
+				head.flash2.forceMove(T)
+				head.flash2 = null
+				head = null
+			to_chat(user, "<span class='notice'>You disassemble the cyborg shell.</span>")
+	else
+		to_chat(user, "<span class='notice'>There is nothing to remove from the endoskeleton.</span>")
+	updateicon()
+
+/obj/item/robot_suit/proc/put_in_hand_or_drop(mob/living/user, obj/item/I) //normal put_in_hands() drops the item ontop of the player, this drops it at the suit's loc
+	if(!user.put_in_hands(I))
+		I.forceMove(drop_location())
+		return FALSE
+	return TRUE
+
+/obj/item/robot_suit/screwdriver_act(mob/living/user, obj/item/I) //Removes the current power cell, or swaps it if you're holding a new one.
+	if(!chest.cell) //If no cell is in the shell
+		if(!user.is_holding_item_of_type(/obj/item/stock_parts/cell)) //And we're also not holding a cell
+			to_chat(user, "<span class='notice'>There doesn't seem to be a power cell to remove from [src].</span>")
+			return
+		else //But we are holding a cell
+			chest.cell = user.held_items[user.active_hand_index == 1 ? 2 : 1]
+			if(!user.transferItemToLoc(user.held_items[user.active_hand_index == 1 ? 2 : 1], src.chest))
+				chest.cell = null
+				to_chat(user, "<span class='warning'>[user.held_items[user.active_hand_index == 1 ? 2 : 1]] is stuck to your hand, you can't put it in [src]!</span>")
+				return
+			to_chat(user, "<span class='notice'>You insert [chest.cell].</span>")
+			return
+	else //If a cell exists in the shell
+		var/obj/item/stock_parts/cell/temp_cell = user.is_holding_item_of_type(/obj/item/stock_parts/cell)
+		if(!temp_cell) //But we're not holding a cell
+			put_in_hand_or_drop(user, chest.cell)
+			to_chat(user, "<span class='notice'>You remove [chest.cell] from [src].</span>")
+			chest.cell = null
+			return
+		else //And we are holding a cell
+			if(!user.transferItemToLoc(temp_cell, chest)) //But we can't insert for some reason
+				put_in_hand_or_drop(user, chest.cell)
+				chest.cell = null
+				to_chat(user, "<span class='warning'>[temp_cell] is stuck to your hand, you can't put it in [src]!</span>")
+				return
+			else //And we put it in
+				put_in_hand_or_drop(user, chest.cell)
+				to_chat(user, "<span class='notice'>You replace [src]'s [chest.cell.name] with [temp_cell].</span>")
+				chest.cell = temp_cell
+				return TRUE

```

Comment:
This can all be condensed down to 
```
	if(QDELETED(chest)) //can't remove a cell if there's no chest to remove it from.
		to_chat(user, "<span class='notice'>There isn't anything to remove from [src]!</span>")
		return

	var/obj/item/stock_parts/cell/temp_cell = user.is_holding_item_of_type(/obj/item/stock_parts/cell)
	var/swap_failed
	if(!temp_cell) //if we're not holding a cell
		swap_failed = TRUE
	else if(!user.transferItemToLoc(temp_cell, chest))
		swap_failed = TRUE
		to_chat(user, "<span class='warning'>[temp_cell] is stuck to your hand, you can't put it in [src]!</span>")

	if(chest.cell) //drop the chest's current cell no matter what.
		put_in_hand_or_drop(user, chest.cell)

	if(swap_failed) //we didn't transfer any new items.
		if(chest.cell) //old cell ejected, nothing inserted.
			to_chat(user, "<span class='notice'>You remove [chest.cell] from [src].</span>")
			chest.cell = null
		return

	to_chat(user, "<span class='notice'>You [chest.cell ? "replace [src]'s [chest.cell.name] with [temp_cell]" : "insert [temp_cell] into [src]"].</span>")
	chest.cell = temp_cell
	return TRUE
```

Same functionality and far easier to read imo.

Reply:
When I went to add the battery insert part, I had re-formatted your code to avoid using extra variables unnecessarily. In hindsight, the nested if statements are probably less preferred. Replacing the screwdriver section.